src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/PropertiesShouldNotBeWriteOnlyTests.cs

```diff
@@ -2,19 +2,845 @@
+        public string CS_AccessibleProperty2
+        {
+            get { return CS_field; }
+            private set { CS_field = CS_value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CS_CA1044Good_protected_Read_private_Write()
+        {
+            var code = @"
+using System;
+namespace CS_GoodPropertiesShouldNotBeWriteOnlyTests3
+{
+    public class CS_ClassWithReadableProperty3
+    {
+        protected string CS_field;
+        protected string CS_AccessibleProperty3
+        {
+        get { return CS_field; }
+        private set { CS_field = CS_value; }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CS_CA1044Good_internal_Read_private_Write()
+        {
+            var code = @"
+using System;
+namespace CS_GoodPropertiesShouldNotBeWriteOnlyTests4
+{
+    public class CS_GoodClassWithReadWriteProperty4
+    {
+        protected string CS_field;
+        internal string CS_AccessibleProperty4
+        {
+            get { return CS_field; }
+            private set { CS_field = CS_value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CS_CA1044Good_protected_internal_Read_internal_Write()
+        {
+            var code = @"
+using System;
+namespace CS_GoodPropertiesShouldNotBeWriteOnlyTests5
+{
+    public class CS_GoodClassWithReadWriteProperty5
+    {
+        protected string CS_field;
+        protected internal string AccessibleProperty5
+        {
+            get { return field; }
+            internal set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CS_CA1044Good_public_Read_internal_Write()
+        {
+            var code = @"
+using System;
+namespace CS_GoodPropertiesShouldNotBeWriteOnlyTests6
+{
+    public class CS_GoodClassWithReadWriteProperty6
+    {
+        protected string CS_field;
+        public string CS_AccessibleProperty6
+        {
+            get { return CS_field; }
+            internal set { CS_field = CS_value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CS_CA1044Good_public_Read_protected_Write()
+        {
+            var code = @"
+using System;
+namespace DesignLibrary
+{
+    public class CS_GoodClassWithReadWriteProperty7
+    {
+        protected string CS_field;
+        public string CS_AccessibleProperty7
+        {
+            get { return CS_field; }
+            protected set { CS_field = CS_value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CS_CA1044Good_public_override_Write()
+        {
+            var code = @"
+using System;
+namespace CS_GoodPropertiesShouldNotBeWriteOnlyTests8
+{
+    protected string CS_field;
+    public class CS_DerivedClassWithReadableProperty : CS_ClassWithReadableProperty
```

Comment:
The definition for the type CS_ClassWithReadableProperty should be part of this test - otherwise test code is erroneous and we are validating on broken code.